content/client-lib-development-guide/features.textile

```diff
@@ -378,14 +383,19 @@ h3(#realtime-channel). Channel
 * @(RTL11)@ If a channel enters the @DETACHED@ or @FAILED@ state, then all messages that are still queued for send on that channel should be deleted from the queue triggering a failure for the publish or presence methods invoked for those messages
 * @(RTL4)@ @Channel#attach@ function:
 ** @(RTL4a)@ If already @ATTACHED@ or @ATTACHING@ nothing is done
+** @(RTL4e)@ If the channel is in the @DETACHING@ state, it will result in an error, as the channel must complete the detach request before reattaching
```

Comment:
I'm not sure either of these is necessary. The `ATTACH` message can be sent before receiving a `DETACHED`.

The client state is:

- set to `ATTACHED`, `DETACHED`, or `FAILED` on receipt of the corresponding message;
- set to `ATTACHING` or `DETACHING` after sending the corresponding request.

An error is definitely bad because the library should be attempting always to relieve the developer of explicitly managing the asynchronous nature of connection and channel state transitions (eg by queueing messages).